javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java

```diff
@@ -34,11 +37,19 @@ public void beforeAgent(Config config) {
   @SuppressWarnings("unused")
   public static synchronized void installAgentTracer(Config config) {
     if (config.getBooleanProperty(JAVAAGENT_ENABLED_CONFIG, true)) {
+
+      System.setProperty("io.opentelemetry.context.contextStorageProvider", "default");
```

Comment:
Adding a dependency on `io.opentelemetry:opentelemetry-extension-noop-api` resulted in `NoopContextStorageProvider` being used instead of the default provider. 
I know it's not the best way to fix it, but it's the only way I could make that work so far.

Reply:
@jkwatson Is this side-effect expected?

Reply:
@anuraaga might be a better person to answer this, but the intent with the true no-op API/SDK was indeed to have it be a true no-op, so the context would also be a no-op. Whether that's desired here or not is a separate question, I guess.

Reply:
@Enkelian Why do want default context storage with noop API/SDK then? :) Do you want "no-op" or "some-op"? :)

Reply:
Default configuration of context storage is used when noop is not enabled (see current version, I've moved this line after @mateuszrzeszutek review). This line is necessary because the presence of noop extension dependency in gradle resulted in `NoopContextStorageProvider` being used in every case - even when noop configuration was not being set up.

Reply:
> the presence of noop extension dependency in gradle resulted in `NoopContextStorageProvider` being used in every case - even when noop configuration was not being set up.

A! That sounds like a bug. Or at least an unintended side effect that needs fixing.

Reply:
Yeah we might need that.

Just for reference, the noop API artifact is still alpha precisely in case we need to make any changes to support its use cases so we can make any fixes we need to it.